core/state/state_object.go

```diff
@@ -283,8 +310,13 @@ func (s *stateObject) setState(key, value common.Hash) {
 // finalise moves all dirty storage slots into the pending area to be hashed or
 // committed later. It is invoked at the end of every transaction.
 func (s *stateObject) finalise() {
+	trieChanges := make([]common.Hash, 0, len(s.dirtyStorage))
 	for key, value := range s.dirtyStorage {
 		s.pendingStorage[key] = value
+		trieChanges = append(trieChanges, key)
+	}
+	if len(trieChanges) > 0 && s.db.prefetcher != nil && s.data.Root != emptyRoot {
+		s.db.prefetcher.PrefetchStorage(s.data.Root, trieChanges)
```

Comment:
Prehaps we can cut down the number of slots to be resolved.

If the original value in storage is empty, then we don't need to resolve the path of slots.

But no idea how many this kind of slots we will have(create new slots in the contract).

Reply:
Why don't we need to resolve the path is the original value is empty? if we're going to write to it, we'll still need to resolve the path.